projects/map-ol/src/lib/map-ol.service.ts

```diff
@@ -897,6 +902,44 @@ export class MapOlService {
     }
   }
 
+  private create_wfs_layer(l: VectorLayer): olVectorLayer {
+
+    // Update URL with current projection
+    const url = getUrlFromUri(l.url);
+    const paras = getParameterJsonFromUri(l.url);
+    // The below re-projection of the bbox is not required, because in WFS (contrary to WMS)
+    // a bbox ('BBOX') may have another projection than the output-projection ('SRSNAME')
+    // if (paras['SRSNAME'] !== this.EPSG) {
+    //   if (paras['BBOX']) {
+    //     this.reprojectFeatures()
+    //   }
+    // }
+    paras['SRSNAME'] = this.EPSG;
+    let fullUrl = `${url}?`;
+    for (const para in paras) {
+      if (paras[para]) {
+        fullUrl += `&${para}=${paras[para]}`;
+      }
+    }
+
+
+    const wfsSource = new olVectorSource({
+      format: new olGeoJSON(),
+      url: fullUrl
+    });
+
+    const styling = l.options?.style || undefined;
+
+    const wfs = new olVectorLayer({
```

Comment:
We should create the olLayer with the `Layer`.options e.g. like on the `wms layer`


```ts

const layeroptions: any = {
        type: 'wfs',
        filtertype: l.filtertype,
        name: l.name,
        id: l.id,
        visible: l.visible,
        legendImg: l.legendImg,
        opacity: l.opacity || 1,
        zIndex: 1,
        source: olSource
      };

new olVectorLayer(layeroptions)
```

Then you also don't need to set the id later.

Reply:
Done!